### DISCUSSION TITLE
Google Cloud, FreeDNS , API secret - Bizarre issue with 2 sites/2 kiddos

### DISCUSSION THREAD

**Discussion by myersc** [NightscoutFoundation/xDrip#2614] [Q&A]
Firstly, not a programmer - thank you to all of you that are amazing programmers and amazingly give of your time to others like me.
Secondly, as I'm not a programmer, I don't do this often and am not used to checking github, I'll be looking to fix this, but I don't do this everyday, so I've tried to include what I think are the relevant details, but I might have missed some, so thanks for your patience.

Setup first NS site for daughter on Google Cloud following [Navid's Github Site](https://navid200.github.io/xDrip/docs/Nightscout/GoogleCloud?fbclid=IwAR1eQkWLxzVOxxxmfkotuIPIcoGIQtLfSBxmLyGp_ixYRJYwvSk2r3Of6eM) with the exception of xDrip - used Loop to upload instead and installed Shuggah as a backup.  Everything was looking beautiful - carbs, basals, boluses!

Setup second NS site for son on Google Cloud (to keep it free?), same as above with the following exceptions:
- Different Google account for Google Cloud
- Different api secret
- Different subdomain
- Uploading via Shuggah (not Looping yet)

It worked great too.  Then I wanted to see nightscout sites side by side....

When daughter's NS loaded it asked for the api secret again (though I'd saved it).  I put in her api and it said it was wrong.  Entered it a couple times and each time wrong.  Input the api for son's nightscout on daughter's NS site and it worked.... only now daughter's NS now displayed son's glucose data.

Thinking maybe I'd used the wrong Google account and rebuilt daughter's NS, I went back to Google account for daughter's NS and completed “Installation Phase 2” for daughter's Google cloud and re-entered correct api secret (even though it was listed correctly) and finished “Installation Phase 2”.  On loading daughter's NS, all is well, correct glucose data, boluses, etc....   On reloading son's NS it asks for the api secret, his api secret is "incorrect" and inputting daughter's api secret into son's NS gives "correct" result, loads NS with daughter's glucose data.

Also, just as a curiosity question - my previous NS didn't require the api secret to be entered to view glucose data, but now it makes me input that before it loads the site.  Is this the default?  Can I change that?

As mentioned above
Google Cloud x2, FreeDNS (public domain) x2 subdomains, different apis, daughter uploading via Loop, son uploading via Shuggah
Sites:
z.snapchut.com
kately.snapchut.com

TL:DR
Two kids, two different google cloud accounts, two different subdomains, two different api secrets.  Creating second NS changes api and glucose data for first NS site.  Rebuilding first NS changes api and glucose data for both sites.

posted to [cgm in the cloud FB](https://www.facebook.com/groups/cgminthecloud) (fewer details)

**Comment by Navid200** [comment]
Please remove the api-secret from the post above.  otherwise, anyone will be able to log in and make edits.

Please open a terminal using ssh for one of the account and post a screenshot of the status page.

**Reply by Navid200** [reply]
When you go to the status page, you can choose to see the hostname and password.
Can you check those on both virtual machines?
Are they different?

**Reply by myersc** [reply]
The hostname and password are different for each and are appropriate for their respective sites.

![image](https://user-images.githubusercontent.com/30415269/211242678-5d52d55e-3c85-4d17-a80c-e95dcf56047d.png)

![image](https://user-images.githubusercontent.com/30415269/211242432-cd6ef5a7-fb5e-4727-b6fe-2fe0e7879b84.png)


(again noob here, but I'm curious - assuming someone had the API secret, couldn't I just run “Installation Phase 2” on the virtual machine and it would all be good again?)

**Reply by Navid200** [reply]
First, the answer to your last question:  You can change the api_secret by just editing your Nightscout variables if that's what you mean.  So, you won't even need to rerun installation phase 2.

But, I won't even share my hostname with anyone.  
But, I think I understand you that after everything is sorted out, you can change the password.  That is correct.  But, even right now while we are trying to fix it, someone can log in and change something intentionally or unintentionally.  Why would they do that?  I don't know.
Just, please don't share the password.

Now, looking at the two screenshots, you can see that first one has problems as you can see in red color text.

Can I ask you to create a third subdomain?
Then, run phase 2 on the virtual machine that has the red letters.  Enter the new subdomain.
Please do not swap passwords.  You may think that you are troubleshooting.  But, please don't do that.  Please follow the instructions.  if they fail, please tell me.  please don't use the other password or the other hostname.

**Reply by Navid200** [reply]
When you run phase 2, it asks you to enter your FreeDNS user ID and password.
After you do that, it should tell you that you have more than one subdomain and ask you to identify which subdomain you want to use.
Did it do that?

What you describe suggests that you keep entering the same subdomain every time.

It is getting the time for me to go to bed now.  Past midnight here.
I will help you resolve this.  May be not right away.  But, as soon as possible.
Please do what I suggested and I will get back and respond to you in about 8 hours from now.

**Comment by myersc** [comment]
New subdomain, on (formerly) mismatched virtual machine

![image](https://user-images.githubusercontent.com/30415269/211244960-242b994b-f8c9-4da2-8a35-d1ee8fcc8dad.png)

I'm not seeing glucose data yet, but I'll let it rest overnight (and you as well).
Thanks

**Reply by Navid200** [reply]
But, where you able to log in and authenticate?
Is the other one still working?

**Reply by Navid200** [reply]
You asked a question about why you had to enter the password when you didn't in the past.   This page should answer that question:  https://navid200.github.io/xDrip/docs/Nightscout/Auth_Default_Roles.html

**Reply by myersc** [reply]
yes, I was able to authenticate
the other one is still working

thanks for that page

**Reply by Navid200** [reply]
If the password is Apple and if the hostname is bob@web.com, you set up a master, the app that you want to upload to Nighscout, this way:

https://Apple@bob.web.com/api/v1/

That is the base URL you need to enter into the app to use to connect to Nightscout and upload.
If the app that is supposed to upload does not work, may be others who use the same app can tell you if it has some special setup requirement.
